### ISSUE TITLE
BigGAN: progressive growing as stability/performance optimization to reach 1024px

### ISSUE THREAD

**Issue by gwern** [tensorfork/tensorfork#17]
Progressive growing of GANs is one of just a few methods for reaching 1024px high-quality images. Training 1024px GANs from scratch due to the size of the model, enormous number of pixels, and low throughput, is highly unstable and slow. We would like to, say, train 128px BigGAN models for many epoches, and then upscale to 256px, then 512px, then 1024px. The last two phases will be especially slow, but progressive growing lets us skip most of the time at them, since they would only need limited learning of upscalingness.

Thus, a strategy for doing 1024px [Ganbooru](https://github.com/tensorfork/tensorfork/issues/12) would be to take our current [Evonorm](https://github.com/tensorfork/tensorfork/issues/8)/flood/[SimCLR](https://github.com/tensorfork/tensorfork/issues/11) BigGAN which has been showing good results at 256px, try switching to [BigGAN-deep](https://github.com/tensorfork/tensorfork/issues/13) for efficiency, add self-attention at 64px and 128px or 256px, use [gradient checkpointing](https://github.com/tensorfork/tensorfork/issues/15) to avoid OOMing from the additional self-attention memory use, use [gradient accumulation](https://github.com/tensorfork/tensorfork/issues/16) to fake large _n_=4096 minibatches while actually having only _n_=512, and using progressive growing to do most training at lower resolutions like 128px and spending less time at the large inefficient resolutions.
